test/suites/metrics.sh

```diff
@@ -0,0 +1,31 @@
+test_metrics() {
+  ensure_import_testimage
+  ensure_has_localhost_remote "${LXD_ADDR}"
+
+  lxc launch testimage c1
+  lxc init testimage c2
+
+  # c1 metrics should show as the container is running
+  lxc query "/1.0/metrics" | grep "name=\"c1\""
+
+  # c2 metrics should not exist as it's not running
+  ! lxc query "/1.0/metrics" | grep "name=\"c2\"" || false
+
+  # create new certificate
+  openssl req -x509 -newkey rsa:2048 -keyout "${TEST_DIR}/metrics.key" -nodes -out "${TEST_DIR}/metrics.crt" -subj "/CN=lxd.local"
+
+  # this should fail as the certificate is not trusted yet
+  curl -k -s --cert "${TEST_DIR}/metrics.crt" --key "${TEST_DIR}/metrics.key" -X GET "https://${LXD_ADDR}/1.0/metrics" | grep "not authorized"
```

Comment:
Does curl return a non-zero exit code when getting  non-200 status code, as you could then just use `! curl ... || false`?

Reply:
It actually returns 0, yes. I was surprised by that as well.

Reply:
Does something like this work? - change 200 to the expected error code

```
[ $(curl ... -o /dev/stderr -w "%{http_code}") -eq 200 ]
```

Reply:
This should work by the way and would be less error prone than grep:

```
curl https://www.linuxcontainers.org -s -o /dev/null -w "%{http_code}"
301
```

Just need to suppress the progress output, see https://superuser.com/questions/272265/getting-curl-to-output-http-status-code

Reply:
It does work locally. It just doesn't work on Jenkins. It returns "000".